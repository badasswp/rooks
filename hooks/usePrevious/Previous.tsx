import { useState } from 'react';
import type { JSX } from 'react';
import usePrevious from './';

/**
 * Previous component.
 *
 * This component is used to demonstrate
 * the use of the usePrevious hook.
 *
 * @returns {JSX.Element} The Previous component.
 */
const Previous = (): JSX.Element => {
  const [count, setCount] = useState(0);
  const prevCount = usePrevious(count);

  return (
    <div>
      <p data-testid="current">Current: {count}</p>
      <p data-testid="previous">Previous: {prevCount}</p>
      <button onClick={() => setCount(count + 1)}>Increment</button>
    </div>
  );
}

export default Previous;
