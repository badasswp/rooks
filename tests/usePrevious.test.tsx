import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import Previous from '../hooks/usePrevious/Previous';

describe('<UsePrevious />', () => {
  it('should show the component with no rendering issues', () => {
    const { container } = render(<Previous/>);

    expect(container).toMatchSnapshot();
  });
});
