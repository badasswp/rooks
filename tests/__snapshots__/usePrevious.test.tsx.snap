// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsePrevious /> should return a sanitized string 1`] = `
<div>
  <div>
    <p>
      Current: 
      0
    </p>
    <p>
      Previous: 
      0
    </p>
    <button>
      Increment
    </button>
  </div>
</div>
`;

exports[`<UsePrevious /> should show the component with no rendering issues 1`] = `
<div>
  <div>
    <p>
      Current: 
      0
    </p>
    <p>
      Previous: 
      0
    </p>
    <button>
      Increment
    </button>
  </div>
</div>
`;
